marklogic-client-api/src/main/java/com/marklogic/client/impl/RESTServices.java

```diff
@@ -450,37 +464,116 @@ public SingleAtomicCallField(String paramName, String paramValue) {
     private Stream<String> paramValues;
-    public MultipleAtomicCallField(String paramName, Stream<String> paramValues) {
+    public UnbufferedMultipleAtomicCallField(String paramName, Stream<String> paramValues) {
       super(paramName);
       this.paramValues = paramValues;
     }
-    public Stream<String> getParamValues() {
-      return paramValues;
+    
+    @Override
+    public BufferedMultipleAtomicCallField toBuffered() {
+    	return new BufferedMultipleAtomicCallField(super.paramName, paramValues);
     }
+
+    @Override
+    public Stream<String> getParamValues() {
+		return paramValues;
+	}
   }
-  static public class SingleNodeCallField extends CallField {
+  static public class UnbufferedSingleNodeCallField extends CallField implements SingleNodeCallField {
     private AbstractWriteHandle paramValue;
-    public SingleNodeCallField(String paramName, AbstractWriteHandle paramValue) {
+    public UnbufferedSingleNodeCallField(String paramName, AbstractWriteHandle paramValue) {
       super(paramName);
       this.paramValue = paramValue;
     }
+    @Override
     public AbstractWriteHandle getParamValue() {
       return paramValue;
     }
+    @Override
+    public BufferedSingleNodeCallField toBuffered() {
+    	if(!(paramValue instanceof BufferableHandle))
+    		throw new IllegalArgumentException("Default parameter must implement BufferableHandle.");
+    	BytesHandle bytesHandle = new BytesHandle((BufferableHandle) paramValue);
+    	return new BufferedSingleNodeCallField(super.paramName, Stream.of(bytesHandle));
+    }
+
   }
-  static public class MultipleNodeCallField extends CallField {
+  static public class UnbufferedMultipleNodeCallField extends CallField implements MultipleNodeCallField {
     private Stream<? extends AbstractWriteHandle> paramValues;
 
-    public MultipleNodeCallField(String paramName, Stream<? extends AbstractWriteHandle> paramValues) {
+    public UnbufferedMultipleNodeCallField(String paramName, Stream<? extends AbstractWriteHandle> paramValues) {
       super(paramName);
       this.paramValues = paramValues;
     }
+    
+    @Override
     public Stream<? extends AbstractWriteHandle> getParamValues() {
       return paramValues;
     }
+    
+    @Override
+    public BufferedMultipleNodeCallField toBuffered() {
+    	if(!(paramValues instanceof BufferableHandle))
+    		throw new IllegalArgumentException("Default parameter must implement BufferableHandle.");
+    	BytesHandle bytesHandle = new BytesHandle((BufferableHandle) paramValues);
+    	return new BufferedMultipleNodeCallField(super.paramName, Stream.of(bytesHandle));
+    }
+    
+  }
+  static public class BufferedMultipleAtomicCallField extends CallField implements MultipleAtomicCallField {
+	  
+	private String[] paramValues;
+	public BufferedMultipleAtomicCallField(String paramName, Stream<String> paramValues) {
+		super(paramName);
+        this.paramValues = (String[]) paramValues.toArray();
```

Comment:
My understanding is that the preferred way to construct the array is using the lambda, as in:

    `paramValues.toArray(size -> new String[size])`

That way, the array is constructed once with the correct type and the correct number of elements for filling by the stream.